web/controllers/user_controller.ex

```diff
@@ -11,6 +11,22 @@ defmodule CoursePlanner.UserController do
     render(conn, "index.html", users: Users.all())
   end
 
+  def show(%{assigns: %{current_user: %{role: role}}} = conn, %{"id" => id}) do
+    if role === "Coordinator" do
```

Comment:
I could deal with the pattern matching, but I couldn't solve the issue, so now every user can see their own show page. Gui created a card for cleaning up permissions